swift-sdk/Internal/IterableAPIInternal.swift

```diff
@@ -338,11 +338,10 @@ final class IterableAPIInternal: NSObject, PushTrackerProtocol, AuthProvider {
                                  forResult: apiClient.inAppConsume(messageId: messageId))
     }
     
-    func inAppConsume(message: IterableInAppMessage, location: InAppLocation, source: InAppDeleteSource) {
+    func inAppConsume(message: IterableInAppMessage, location: InAppLocation = .unknown, source: InAppDeleteSource = .unknown) {
```

Comment:
Does unknown actually get passed to the backend?

Currently the backend only expects in-app or inbox, so the api request might fail on this. https://github.com/Iterable/Iterable/blob/master/app/models/api/InboxMessageContext.scala#L44
Is there a use case for not passing in this location?

If we want to accept no value for the location, then we will need to change the backend to support a non-existent value.

Reply:
No unknown does not get passed to the backend. If the value is unknown we don't sent any value to the backend.

Reply:
In the api contract location is required, so the api call would fail if the location value is not there. Sounds like we would need a change to the api to support this.

Reply:
We will not always know the location. I can pass a default location but I think that is not ideal.

Reply:
Plus previous versions of SDK will not have location. So that needs to be optional anyway.

Reply:
Technically it adheres to backwards since the messageContext object is optional. 
I'll go ahead an make location optional as well.

Reply:
Ah. I see.

Reply:
In that case.... I am also OK providing a default location.

Reply:
I have created a separate Jira for "unknown/other". https://iterable.atlassian.net/browse/MOB-748